File: queue/queue.py
class Node:
  def __init__(self, val, next=None):
    self.val = val
    self.next = next


class Queue:
  def __init__(self, front=None, back=None):
    self.front = front
    self.back = back
  def push(self, val):
    #add element to the end of the ll
    new_node = Node(val)
    if self.back == None:
      self.front = new_node
      self.back = new_node  
    else:
      new_node.next = self.back 
      self.back = new_node 
    
  def pop(self):
    # remove first element from the ll
    pass
  def peek(self):
    #return the first element
    pass

File: queue.c
#include<stdio.h>

typedef struct node{
    int data;
    struct node *link;
}Node;

Node *front=NULL;
Node *rear=NULL;

void enqueue(int x){
    Node *newnode;
    newnode=(Node*)malloc(sizeof(Node));
    newnode->data=x;
    newnode->link=NULL;
    if(front==NULL && rear==NULL){
        front=rear=newnode;
    }
    else{
        rear->link=newnode;
        rear=newnode;
    }
}

void dequeue(){
    if(front==NULL && rear==NULL){
        printf("Queue is empty");
    }
    else{
    Node *temp;
    temp=front;
    front=front->link;
    free(temp);
    }
}

void peek(){
    if(front==NULL && rear==NULL){
        printf("Queue is empty");
    }
    else{
        printf("First element is : %d",front->data);
    }
}

int main(){
    enqueue(10);
    enqueue(20);
    enqueue(30);
    dequeue();
    peek();
}